tests/func/test_get.py

```diff
@@ -125,9 +126,24 @@ def test_get_full_dvc_path(erepo):
     assert filecmp.cmp(external_data, "ext_data_imported", shallow=False)
 
 
+def test_non_cached_output(request, erepo):
+    os.chdir(erepo.root_dir)
+    erepo.dvc.run(
+        outs_no_cache=["non_cached_file"], cmd="echo hello > non_cached_file"
+    )
+    erepo.dvc.scm.add(["non_cached_file", "non_cached_file.dvc"])
+    erepo.dvc.scm.commit("add non-cached output")
+    new_dir = tempfile.mkdtemp(prefix=request.node.name + "_")
```

Comment:
There is a pytests `tmp_path` fixture, which will create an empty dir for you, which will contain test name in it. It will also clean up automatically.